File: data/src/main/kotlin/dev/yashgarg/kimai/models/TimesheetActivity.kt
/*
 * Copyright © 2023 Yash Garg.
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE file or at
 * https://opensource.org/licenses/MIT.
 */
package dev.yashgarg.kimai.models

import androidx.annotation.Keep
import kotlinx.serialization.Serializable

@Keep
@Serializable
data class TimesheetActivity(
  val activity: Int,
  val project: Int,
  val tags: List<String>,
  val id: Int,
  val begin: String,
  val end: String,
  val duration: Int,
  val description: String? = null,
  val rate: Double,
  val internalRate: Double,
  val exported: Boolean,
  val billable: Boolean,
  val metaFields: List<MetaField>,
)

File: lib/data/models/timesheet_activity.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:kimai/data/models/models.dart';

part '../../generated/data/models/timesheet_activity.freezed.dart';
part '../../generated/data/models/timesheet_activity.g.dart';

@freezed
class TimesheetActivity with _$TimesheetActivity {
  const factory TimesheetActivity({
    required int activity,
    required int project,
    required List<String> tags,
    required int id,
    required String begin,
    required String end,
    required int duration,
    String? description,
    required double rate,
    required double internalRate,
    required bool exported,
    required bool billable,
    required List<MetaField> metaFields,
  }) = _TimesheetActivity;

  factory TimesheetActivity.fromJson(Map<String, dynamic> json) =>
      _$TimesheetActivityFromJson(json);
}
